tests/cpp/test_gpu3.cpp

```diff
@@ -8174,6 +8174,34 @@ TEST_F(NVFuserTest, BlockReduction3D) {
   }
 }
 
+// Simple test to merge an inner domain as an outer input
+TEST_F(NVFuserTest, ReverseMerge) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+  fusion.addOutput(tv1);
+
+  tv1->merge(1, 0);
+
+  ASSERT_EQ(tv1->nDims(), 1);
+  auto merge = dynamic_cast<Merge*>(tv1->axis(0)->definition());
+  ASSERT_NE(merge, nullptr);
+  ASSERT_EQ(merge->outer(), tv1->getRootDomain().at(1));
+  ASSERT_EQ(merge->inner(), tv1->getRootDomain().at(0));
```

Comment:
Nit: these could be EXPECT_EQ

Reply:
It seems to me it'd make little sense to continue this test if this condition is not met, so `ASSERT` seems to make more sense to me.